marshaling_test.go

```diff
@@ -235,6 +291,42 @@ var unmarshalTestCases = []struct {
 			Retryable: nil,
 		},
 	},
+	// Wrapped errors only gets unmarshaled as a single error
```

Comment:
marshaling and unmarshaling are asymmetric to start with, but the error code is preserved.
https://github.com/monzo/terrors/blob/ff4c482e20b3403381f93a06a6192c0ce143ab55/marshaling_test.go#L192